### PR TITLE
Support `create function` CA for workers and explicitly defined anonymous functions

### LINKED ISSUE #42013: [Bug]: Create function code action fails inside workers
### Description

When attempting to apply the create function code action within workers encounters failure.

### Steps to Reproduce

```ballerina
function foo() {
    worker A {
        return <cursor>bar();
    }
}
```

### Affected Version(s)

2201.8.4

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by nipunayf**
## Purpose
The `create function` CA uses the `FunctionCallExpressionTypeFinder`, which determines the expected type of a function call. The implementation of this type finder does not have the logic to handle workers and explicit anonymous functions; thus, these are not supported with the `create function` CA.

Fixes #42013

## Approach
The PR modifies the `FunctionCallExpressionTypeFinder` to support determining the expected type of function calls within workers and explicit anonymous function calls.

## Samples
image
image

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java [hunk 1]
```diff
@@ -402,6 +405,12 @@ public void visit(BlockStatementNode node) {
         node.parent().accept(this);
     }
 
+    @Override
+    public void visit(NamedWorkerDeclarationNode namedWorkerDeclarationNode) {
+        Optional<Symbol> symbol = semanticModel.symbol(namedWorkerDeclarationNode);
+        symbol.ifPresent(value -> checkAndSetTypeResult(((WorkerSymbol) value).returnType()));
```
Comment (LakshanWeerasinghe) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java:411]:
```suggestion
        semanticModel.symbol(namedWorkerDeclarationNode)
            .ifPresent(value -> checkAndSetTypeResult(((WorkerSymbol) value).returnType()));
```

Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java:411]: Apply for other places as well.

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_worker.bal [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    worker B returns string {
+        foo(32);
+        return "bar";
+    }
+
+    worker C {
+        return foo();
+    }
+
+    worker D returns module1:TestRecord2 {
```
Comment (LakshanWeerasinghe) [inline language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_worker.bal:17]: Shall we add a test with a worker inside a fork statement?

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_explicit_anonymous_function.bal [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import ballerina/module1;
+
+function createFunctionInWorker() {
+    var fn1 = function() returns int {
+        return foo();
+    };
+
+    var fn2 = function() returns string {
+        foo(32);
```
Comment (LakshanWeerasinghe) [inline language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_explicit_anonymous_function.bal:9]: Shall we add another test by wrapping with statement inside another lambda function?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java [hunk 4]
```diff
@@ -414,8 +423,6 @@ public void visit(ReturnStatementNode returnStatementNode) {
         if (resultFound && returnTypeSymbol.typeKind() == TypeDescKind.FUNCTION) {
             FunctionTypeSymbol functionTypeSymbol = (FunctionTypeSymbol) returnTypeSymbol;
             functionTypeSymbol.returnTypeDescriptor().ifPresentOrElse(this::checkAndSetTypeResult, this::resetResult);
-        } else {
-            resetResult();
```
Comment (KavinduZoysa) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java:418]: Why is this deleted?

Comment (nipunayf) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java:418]: There are currently two reasons to do this:

1. This resets the type symbol found in the worker node. While the function captures the function type symbol, this cannot be replicated to the worker (as there is no worker type symbol but rather a symbol). This broadens the possible values for the type symbol under the worker node, and hence, we cannot replicate the `returnTypeSymbol.typeKind() == TypeDescKind.FUNCTION` condition for the worker.

2. Looking at the existing logic and test cases, this line appears to be unreachable. AFAIK, the statement is there to reset the results for the negative paths. Currently, this class is only used by the `call function` CA, and it appears to narrow these negative paths, thus making this logic redundant.
